src/Castle.Windsor/MicroKernel/SubSystems/Conversion/TypeNameConverter.cs

```diff
@@ -144,6 +149,29 @@ private bool InitializeAppDomainAssemblies(bool forceLoad)
 					Scan(assembly);
 				}
 			}
+#else
+			if (assemblies.Count == 0)
+			{
+				var context = DependencyContext.Default;
```

Comment:
@fir3pho3nixx did you see my comments in https://github.com/castleproject/Windsor/pull/267#discussion_r119878886?

I've worked on some big .NET applications using Windsor before where the bin directory contained hundreds of DLLs, so eagerly loading all DLLs would massively slow down start up of those applications and increase memory usage. This code path could also be called quite a few times in `TypeNameConverter.GetType` and failed fusion binds throwing exceptions will make things messy.

Alternatively I'm not that impressed with using `Microsoft.Extensions.DependencyModel` either because it is designed for .NET CLI not .NET Core/Standard, it doesn't return the assemblies loaded into the CLR but the "runtime assemblies" specified in the `.deps` file for .NET CLI projects.

I guess our options are limited at this time without going with .NET Standard 2.0.